src/activity/iActivity.inl.hxx

```diff
@@ -2,7 +2,18 @@
 template <typename StateType, typename... Args>
 StateType & iActivity::createState(Args &&... args)
 {
-  StateType* state = new StateType(std::forward(args)..., GetService<UILoopMQ>());
+  StateType* state = new StateType(std::forward<Args>(args)..., GetService<UILoopMQ>());
   m_States.emplace_back(state);
   return *state;
 }
+
+template <typename ControllerType, typename... Args>
+ControllerType & iActivity::createController(Args &&... args)
```

Comment:
The point of this function is to not have to worry about attaching explicitly

Reply:
I wanted to add even more things other than iMouseHandlers to ease the process of creating Controllers

Reply:
Instead of creating one function which will create every possible controller, consider creating multiple functions. With proper naming, it makes way more readable code.

Reply:
It's not one function, it's a function template. We can specialize for any Controller we want if needed. But in the Widget Factory update in Pt II this will be more explicit

Reply:
Perhaps your point is that we can't partially specialize functions, so it makes customization more difficult for multiple Controllers at once. I agree with that. We need Policies to resolve this issue and it will be fixed with widget factories